gen/zap.go

```diff
@@ -26,6 +26,8 @@ import (
 	"go.uber.org/thriftrw/compile"
 )
 
+const NoZapLabel = "go.nolog"
```

Comment:
This should have docs on it. The linter would have complained about it but I
just realized that the linter hasn't been running in CI for a while. Fixing.